src/mappings/entities/mapPortfolio.ts

```diff
@@ -0,0 +1,146 @@
+import { Codec } from '@polkadot/types/types';
+import { SubstrateEvent } from '@subql/types';
+import { Portfolio, Settlement } from '../../types';
+import {
+  getBigIntValue,
+  getNumberValue,
+  getPortfolioValue,
+  getSignerAddress,
+  getTextValue,
+  serializeTicker,
+} from '../util';
+import { EventIdEnum, ModuleIdEnum } from './common';
+import { createLeg, SettlementResultEnum } from './mapSettlement';
+
+export const getPortfolio = async ({
+  identityId,
+  number,
+}: Pick<Portfolio, 'identityId' | 'number'>): Promise<Portfolio> => {
+  const portfolioId = `${identityId}/${number}`;
+
+  let portfolio = await Portfolio.get(portfolioId);
+
+  if (!portfolio) {
+    if (number === 0) {
```

Comment:
What about making a default portfolio when an Identity is created instead?

I think that would make more consistent query results since every Identity should have a portfolio even if its empty.

Reply:
I like the idea. I'll add a default portfolio whenever a new Identity is created